### ISSUE TITLE
Is `templates` important to the final performance ?

### ISSUE THREAD

**Issue by ljch2018** [arielnlee/Platypus#15]
Thank you for sharing the impressive work. 

I try to finetune my own model, but I don't want to use templates in my training prompt.

What's the result after removing it ? Do you test it ?

Thank you.

**Comment by arielnlee** [comment]
Thanks for your interest! We've only used this script for instruction-tuning with the alpaca template (although we are trying orca now as well). You're free to use a template other than alpaca, as long as you add it to the templates folder and make changes to the prompter, if necessary. If you don't want to use instruction-tuning at all, you'll have to edit the script to use whatever formatting you've decided upon. Are you looking to do next token prediction? If you give me additional details, I will give you a more informative approach to editing the scripts.

**Comment by bdambrosio** [comment]
Hi. thanks for this great work! 
Similar question - I'm finetuning on a single fixed instruction followed by a multiple choice question and answer set.
Do you think I should add a template with my single fixed instruction, or would it be better to keep the alpaca template and include my entire instruction and q/a as the 'input'? this is a single-purpose fine-tune, so I'm leaning to defining my own template, but not sure if fine-tuning can 'pull' the model away from the alpaca template it's been trained on. Following is my instruction:
"Following is a multiple choice question, background information, and a set of possible answers.
Analyze the answer choices as they relate to the question. Respond with the letter label for the best answer."

**Comment by lilloukas** [comment]
Are you looking to use the merged platypus model, or the original Llama 2 as the base for your training? If it's the former, I would suggest you keep the alpaca template for any additional fine-tuning. If you are looking at doing a fine-tune over the base Llama 2 model, then modifying the template as follows might be best:
```
{
    "description": "Template for multiple choice questions",
    "prompt_input": "Following is a multiple choice question, background information, and a set of possible answers.
Analyze the answer choices as they relate to the question. Respond with the letter label for the best answer.\n\n### Instruction:\n{instruction}\n\n### Input:\n{input}\n\n### Response:\n",
    "prompt_no_input": "Following is a multiple choice question, background information, and a set of possible answers.
Analyze the answer choices as they relate to the question. Respond with the letter label for the best answer.\n\n### Instruction:\n{instruction}\n\n### Response:\n",
    "response_split": "### Response:"    
}
```
In that case, you should format  your data so that all of the multiple choice q/a pairs are the `instruction`, with the letter answer as the `output`.

 We have seen good results from merges which used a mix of template formats, though we are actively investigating whether (and to what extent) mixed prompts might negatively impact usability.

**Comment by bdambrosio** [comment]
Thanks!
